From efba7fca51a2eac9d2b926f1593e083e97e2bf3a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 02 Jun 2019 10:55:05 +0300
Subject: [PATCH] [SW-4185] gaudi: pll: move div out123 to ref clock

Update the LBW and other divisors to the latest "safe" configuration,
which is the reference clock.
Once we find the W/A needed to raise the PLL outputs' 1,2,3 frequency,
we will be able to restore their settings.

Change-Id: Ib1a9c7e11ae6b0f8f093d24aaed3ff5d11ed7f55
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index d02ddd1..3887b68 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -21,7 +21,7 @@
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
+			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
@@ -31,8 +31,8 @@
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "IF_E", .base = mmIF_E_PLL_BASE,
@@ -40,15 +40,15 @@
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
+			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
@@ -58,8 +58,8 @@
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
@@ -76,8 +76,8 @@
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
@@ -93,9 +93,9 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
@@ -103,25 +103,25 @@
 		.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
 		.div_factor = { 0, 0, 4, 9 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
 		.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
+			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
@@ -130,8 +130,8 @@
 		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
 		.div_factor = { 0, 4, 8, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
 	}, {
@@ -139,8 +139,8 @@
 		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
 		.div_factor = { 0, 4, 8, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
 	}, {
@@ -148,7 +148,7 @@
 		.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
+			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
@@ -157,18 +157,18 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	}, {
 		.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
 		},
 	},
 };
